xchange-core/src/main/java/org/knowm/xchange/currency/Currency.java

```diff
@@ -302,8 +305,9 @@
   public static final Currency STORJ = createCurrency("STORJ", "Storj", null);
   public static final Currency MOD = createCurrency("MOD", "Modum", null);
 
-  private final String code;
-  private final CurrencyAttributes attributes;
+  @JsonProperty private final String code;
```

Comment:
i think this class should serialize to a simple string, not an object, same goes for `CurrencyPair`

when you serialize an order you want it to look like this:

```
{
  "type" : "BID",
  "originalAmount" : "0.234234",
  "currencyPair" : "ETH/BTC",
  // ...
}
```
**NOT like this:**

```
{
  "type" : "BID",
  "originalAmount" : "0.234234",
  "currencyPair" : {
    "base" : {
      "code" : "ETH"
      "attributes" : [
        // tons of metadata that repeat in every file
      ]
    },
    "counter" : {
      "code" : "BTC"
      "attributes" : [
        // tons of metadata that repeat in every file
      ]
    }
  },
  // ...
}
```

I think that just using `@JsonCreator` on `getInstance` and `@JsonValue` on `getCurrencyCode` should do the trick

Reply:
Done